teps/0041-tekton-component-versioning.md

```diff
@@ -0,0 +1,333 @@
+---
+status: proposed
+title: Tekton Component Versioning
+creation-date: "2021-02-01"
+last-updated: "2021-02-01"
+authors:
+  - "@vinamra28"
+  - "@piyush-garg"
+  - "@SM43"
+---
+
+# TEP-0041: Tekton Component Versioning
+
+<!-- toc -->
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+  - [Creating a ConfigMap](#creating-a-configmap)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes to add versioning for all the components of Tekton in
+such a manner that any authenticated user of the cluster on which Pipelines
+is installed can get to know version of any Tekton Component. Also with
+Tekton CLI, we can get to know the version of pipelines and commands like
+install/upgrade/downgrade/reinstall a resource from catalog only if the
+newer resource version is compatible with Pipelines.
+
+## Motivation
+
+Finding the version of tekton components such as Tekton Pipelines etc which
+are currently installed on the cluster is not an easy thing specially when we
+are a non-admin user as we may not have specific permissions to view the
+version. In the current scenario we can fetch version from the controller
+deployment and for that we need:-
```

Comment:
NIT: I think it would be better to be specific, as "admin" may not be well defined. Also, if the required role is available, reading the versions is really straight forward via `tkn version`. I would simply say that what is needed today is the permission to read deployments in the namespace where Tekton is deployed.